app.js

```diff
@@ -13,14 +12,79 @@ app.use(express.static(__dirname + '/public'));
 
 
 // setting the spotify-api goes here:
+const clientId = 'a745883dad764aa78792b9306eaddefd',
+    clientSecret = '5934fadd9a25444f8f2575c82393a288';
 
+const spotifyApi = new SpotifyWebApi({
+  clientId : clientId,
+  clientSecret : clientSecret
+});
 
-
-
+// Retrieve an access token
+spotifyApi.clientCredentialsGrant()
+  .then( data => {
+    spotifyApi.setAccessToken(data.body['access_token']);
+  })
+  .catch(error => {
+    console.log('Something went wrong when retrieving an access token', error);
+  })
 
 
 // the routes go here:
-
+app.get('/',(req,res,next)=>{
+  res.render('index');
+})
+
+
+app.get('/artists',(req,res)=>{
+
+  spotifyApi.searchArtists(req.query.artistName)
+  .then(data => {
+    const artistData = {data}; 
+    // console.log("The received data from the API: ", data.body.artists.items[0]);
+    res.render('artists',{data});
+  })
+  .catch(err => {
+    console.log("The error while searching artists occurred: ", err);
+  })
+})
+
+app.get('/artists/:id/albums',(req,res)=>{
+
+  const {id} = req.params;
+  console.log(id);
+  
+  spotifyApi.getArtistAlbums(id) // passing id in params
+  .then(data => {
+    console.log("The received data from the API: ", data.body);
+    const albums = {data}; 
```

Comment:
You are doing the const albums here but then you pass to the view `albums: data.body`. Try to refactor this